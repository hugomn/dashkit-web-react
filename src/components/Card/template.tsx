import { Box, Heading } from "grommet";
import * as React from "react";
import styled from "styled-components";

interface IProps {
  title: string;
  buttons?: React.ReactNode;
}

const Card: React.FC<IProps> = ({ children, title, buttons }) => {
  return (
    <Box margin={{ top: "medium" }} background="light-1" elevation="xsmall" pad="medium">
      <Box direction="row" align="center" justify="between" margin={{ bottom: "small" }}>
        <StyledHeading level="5">{title}</StyledHeading>
        {buttons}
      </Box>
      <Box direction="row" align="center">
        {children}
      </Box>
    </Box>
  );
};

const StyledHeading = styled(Heading)`
  margin: 0;
`;

export default Card;
