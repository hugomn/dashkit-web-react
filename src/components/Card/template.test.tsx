import { render } from "@testing-library/react";
import * as React from "react";
import Card from "./template";

describe("Card component", () => {
  it("Should match snapshot", () => {
    const { container } = render(<Card title="title" />);
    expect(container).toMatchSnapshot();
  });
});
