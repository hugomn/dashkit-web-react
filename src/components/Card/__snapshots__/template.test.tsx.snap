// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Should match snapshot 1`] = `
<div>
  <div
    class="StyledBox-sc-13pk1d4-0 kneDRP"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 gjKiFz"
    >
      <h5
        class="sc-bdVaJa bcSwbK StyledHeading-sc-1rdh4aw-0 jDZxiw"
      >
        title
      </h5>
    </div>
    <div
      class="StyledBox-sc-13pk1d4-0 gQLnBt"
    />
  </div>
</div>
`;
